#! /bin/sh /usr/share/dpatch/dpatch-run
## 02_configure.dpatch by Torsten Marek <shlomme@debian.org>
##
## DP: patch the configure script for the Python bindings to support object dir builds

@DPATCH@
Index: qscintilla2-2.4.2/Python/configure.py
===================================================================
--- qscintilla2-2.4.2.orig/Python/configure.py	2010-01-20 20:56:09.000000000 +0100
+++ qscintilla2-2.4.2/Python/configure.py	2010-01-27 22:54:35.603870417 +0100
@@ -33,6 +33,7 @@
 import glob
 import optparse
 
+src_dir = os.path.dirname(os.path.abspath(__file__))
 
 # Import SIP's configuration module so that we have access to the error
 # reporting.  Then try and import the configuration modules for both PyQt3 and
@@ -162,7 +163,7 @@
     if os.access(sciglobal, os.F_OK):
         # Get the QScintilla version string.
         _, sciversstr = sipconfig.read_version(sciglobal, "QScintilla", "QSCINTILLA_VERSION", "QSCINTILLA_VERSION_STR")
-
+        return # Debian: do not check for the installed version, we're good this way.
         if glob.glob(os.path.join(opts.qscilibdir, "*qscintilla2*")):
             # Because we include the Python bindings with the C++ code we can
             # reasonably force the same version to be used and not bother about
@@ -232,9 +233,9 @@
     argv.append(buildfile)
 
     if pyqt.pyqt_version >= 0x040000:
-        argv.append("sip/qscimod4.sip")
+        argv.append(os.path.join(src_dir, "sip/qscimod4.sip"))
     else:
-        argv.append("sip/qscimod3.sip")
+        argv.append(os.path.join(src_dir, "sip/qscimod3.sip"))
 
     os.system(" ".join(argv))
 
